kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/TypeResolver.kt

```diff
@@ -35,18 +35,18 @@ open class TypeResolver {
 
       is ParameterizedTypeName -> {
             typeName.rawType.parameterizedBy(*(typeName.typeArguments.map { resolve(it) }.toTypedArray()))
-            .asNullableIf(typeName.nullable)
+            .asNullableIf(typeName.isNullable)
       }
 
       is WildcardTypeName -> {
         when {
-          typeName.lowerBounds.size == 1 -> {
-            WildcardTypeName.supertypeOf(resolve(typeName.lowerBounds[0]))
-                .asNullableIf(typeName.nullable)
+          typeName.inTypes.size == 1 -> {
+            WildcardTypeName.consumerOf(resolve(typeName.inTypes[0]))
```

Comment:
random thought - would it make more sense for these to be symmetric?

```kotlin
WildcardTypeName.inTypesOf(...)
WildcardTypeName.outTypesOf(...)
```

You can nicely import them as a top-level function that way too, like `listOf()`

Reply:
I think it'd be `ofOutTypes()` and `ofInTypes()`.